### PR TITLE
[TypeScript Fetch] Optional URL query parameters not sent anymore

### LINKED ISSUE #6003: [TypeScript Fetch] Optional URL query parameters are sent empty if not present
##### Description
When an URL parameter is optional and **not** specified it results in requests that can break back-end systems.

Generated stub, correctly specifying that `v` and `open` are optional
```
getRouteUsingGET(params: {  "src": string; "dst": string; "open"?: string; "v"?: string; }, options?: any): FetchArgs {
```

Hoewver, using it like this
```
SomeApi.RouteApiFactory().getRouteUsingGET({
            src: "foo",
            dst: "bar",
            open: "PAN"
            // no v
        })
```
results in a requested URL that looks like `http://localhost:8080/someapi/route/foo/bar?open=PAN?v=`.
While this might strictly be standard compliant it can break back-end frameworks. Spring for example will think the `open` parameter is `PAN?v=` instead of it being `PAN`.

##### Swagger-codegen version
2.2.2

##### Swagger declaration file content or url

```
/distance/route/{srcPort}/{dstPort}: {
get: {
consumes: [
"application/json"
],
produces: [
"*/*"
],
parameters: [
{
name: "open",
in: "query",
required: false,
type: "string"
},
{
name: "v",
in: "query",
required: false,
type: "string"
}
],
responses: {...}
...
}
```
##### Command line used for generation

`java -jar swagger-codegen-cli.jar generate -i http://localhost:8090/v2/api-docs -l typescript-fetch -o output/`

##### Suggest a Fix

Don't send empty query parameters.

### PR DESCRIPTION
**Pull request by roberterdin**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR
This prevents empty URL parameters from being sent, e.g. `http://some-api/endpoint?param=SOMETHING?v=` will now be `http://some-api/endpoint?param=SOMETHING`.
Solution suggested by @TiFu. Fixes #6003.

Can somebody please verify whether this is still working as intended: https://github.com/swagger-api/swagger-codegen/compare/master...roberterdin:master#diff-89eaac82e75bf2e3698813ad62b26691L70
I don't know if it is covered by the tests and if the browser tests are being executed with maven.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/swagger-api/swagger-codegen/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
cc @Vrolijkx @taxpon @sebastianhaas @kenisteward @TiFu

**Comment by roberterdin** [comment]
Thanks for reminding me to update the email address (my employer managed to have 4 email addresses for every employee :wink:)

File: modules/swagger-codegen/src/main/resources/TypeScript-Fetch/api.mustache [hunk 1]
```diff
@@ -90,11 +90,12 @@ export const {{classname}}FetchParamCreator = {
             .replace(`{${"{{baseName}}"}}`, `${ params["{{paramName}}"] }`){{/pathParams}};
         let urlObj = url.parse(baseUrl, true);
 {{#hasQueryParams}}
-        urlObj.query = {{#supportsES6}}Object.{{/supportsES6}}assign({}, urlObj.query, {
-            {{#queryParams}}
-            "{{baseName}}": params["{{paramName}}"],
-            {{/queryParams}}
-        });
+    urlObj.query =  {{#supportsES6}}Object.{{/supportsES6}}assign({}, urlObj.query)
```
Comment (TiFu) [inline modules/swagger-codegen/src/main/resources/TypeScript-Fetch/api.mustache]: Could you please fix the indentation of urlObj.query and then regenerate the samples? Thanks :-)

Comment (roberterdin) [inline modules/swagger-codegen/src/main/resources/TypeScript-Fetch/api.mustache]: Fixed it... tell me if you want me to squash the commits or whether you rebase anyway.
Cheers

**Comment by wing328** [comment]
The TS Fetch Petstore tests failed with the following errors reported by Travis CI:
```
npm WARN typescript-fetch-api@0.0.0 No repository field.
> typescript-fetch-api@0.0.0 test /Users/travis/build/swagger-api/swagger-codegen/samples/client/petstore/typescript-fetch/builds/default
> tslint api.ts
api.ts[151, 49]: Missing semicolon
api.ts[174, 49]: Missing semicolon
api.ts[975, 49]: Missing semicolon
npm ERR! Test failed.  See above for more details.
```

Please take a look when you've time.

**Comment by wing328** [comment]
The AppVeyor build errors can be ignored.

**Comment by wing328** [comment]
Github rate limit again:
```
-> running reinstall

[ERR!] GitHub API: GitHub rate limit reached.

    To increase the limit use GitHub authentication.

    See: https://github.com/DefinitelyTyped/tsd#tsdrc
```
Will restart the job later.

**Comment by wing328** [comment]
Tests passed via https://travis-ci.org/swagger-api/swagger-codegen/builds/256418508

**Comment by wing328** [comment]
@roberterdin thanks for the PR, which has been merged into master.

**Pull request merged**